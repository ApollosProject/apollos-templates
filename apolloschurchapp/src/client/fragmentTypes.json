{"__schema":{"types":[{"kind":"INTERFACE","name":"Node","possibleTypes":[{"name":"ContentChannel"},{"name":"Person"},{"name":"Campus"},{"name":"Event"},{"name":"Group"},{"name":"Scripture"},{"name":"Pass"},{"name":"UniversalContentItem"},{"name":"DevotionalContentItem"},{"name":"MediaContentItem"},{"name":"ContentSeriesContentItem"},{"name":"WeekendContentItem"},{"name":"ActionListFeature"},{"name":"TextFeature"},{"name":"ScriptureFeature"}]},{"kind":"INTERFACE","name":"ContentItem","possibleTypes":[{"name":"UniversalContentItem"},{"name":"DevotionalContentItem"},{"name":"MediaContentItem"},{"name":"ContentSeriesContentItem"},{"name":"WeekendContentItem"}]},{"kind":"INTERFACE","name":"Media","possibleTypes":[{"name":"ImageMedia"},{"name":"VideoMedia"},{"name":"AudioMedia"}]},{"kind":"INTERFACE","name":"MediaSource","possibleTypes":[{"name":"ImageMediaSource"},{"name":"VideoMediaSource"},{"name":"AudioMediaSource"}]},{"kind":"INTERFACE","name":"Sharable","possibleTypes":[{"name":"SharableContentItem"},{"name":"SharableFeature"}]},{"kind":"INTERFACE","name":"Feature","possibleTypes":[{"name":"ActionListFeature"},{"name":"TextFeature"},{"name":"ScriptureFeature"}]}]}}