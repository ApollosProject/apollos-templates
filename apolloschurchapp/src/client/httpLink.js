import { Platform } from 'react-native';
import { createUploadLink } from 'apollo-upload-client';
import ApollosConfig from '@apollosproject/config';
import { split } from 'apollo-link';
import { createHttpLink } from 'apollo-link-http';
import { getMainDefinition } from 'apollo-utilities';

let uri = ApollosConfig.APP_DATA_URL;
const androidUri = ApollosConfig.ANDROID_URL || '10.0.2.2';

// Android's emulator requires localhost network traffic to go through 10.0.2.2
if (Platform.OS === 'android') uri = uri.replace('localhost', androidUri);

export default split(
  ({ query }) => {
    const { kind, operation } = getMainDefinition(query);
    return kind === 'OperationDefinition' && operation === 'mutation';
  },
  createUploadLink({ uri }),
  createHttpLink({
    uri,
    useGETForQueries: true,
    headers: {
      // We can safely send these headers.
      // Fastly does not currently respect no-store or no-cache directives. Including either or both of these in a Cache-Control header has no effect on Fastly's caching decision
      // https://docs.fastly.com/en/guides/configuring-caching#do-not-cache
      'Cache-Control': 'no-cache, no-store',
    },
  })
);
