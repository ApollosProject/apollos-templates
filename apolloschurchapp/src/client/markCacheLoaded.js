import gql from 'graphql-tag';

export default gql`
  mutation markCacheLoaded {
    cacheMarkLoaded @client
  }
`;
