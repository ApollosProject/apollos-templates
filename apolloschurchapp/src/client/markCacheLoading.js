import gql from 'graphql-tag';

export default gql`
  mutation markCacheLoading {
    cacheMarkLoading @client
  }
`;
