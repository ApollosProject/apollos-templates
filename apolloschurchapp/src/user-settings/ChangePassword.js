import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import {
  Platform,
  KeyboardAvoidingView,
  StyleSheet,
  StatusBar,
} from 'react-native';
import { Mutation } from '@apollo/client/react/components';
import { Header } from '@react-navigation/stack';
import { Formik } from 'formik';
import * as Yup from 'yup';

import {
  Button,
  TextInput,
  PaddedView,
  FlexedView,
  styled,
} from '@apollosproject/ui-kit';

import { SafeAreaView } from 'react-native-safe-area-context';
import GET_AUTH_TOKEN from '../store/getAuthToken';
import CHANGE_PASSWORD from './passwordChange';

const Footer = styled({
  flex: 1,
  justifyContent: 'flex-end',
})(SafeAreaView);

const StyledKeyboardAvoidingView = styled(({ theme }) => ({
  ...StyleSheet.absoluteFill,
  backgroundColor: theme.colors.background.paper,
}))(KeyboardAvoidingView);

class ChangePassword extends PureComponent {
  static propTypes = {
    navigation: PropTypes.shape({
      navigate: PropTypes.func,
      goBack: PropTypes.func,
    }),
  };

  renderForm = (props) => (
    <StyledKeyboardAvoidingView
      behavior={'padding'}
      keyboardVerticalOffset={
        Header.HEIGHT +
        (Platform.OS === 'android' ? StatusBar.currentHeight : 0)
      }
    >
      <FlexedView>
        <PaddedView>
          <TextInput
            label="New Password"
            type="password"
            value={props.values.password}
            error={props.touched.password && props.errors.password}
            onChangeText={(text) => props.setFieldValue('password', text)}
          />
          <TextInput
            label="Confirm Password"
            type="password"
            value={props.values.confirmPassword}
            error={
              props.touched.confirmPassword && props.errors.confirmPassword
            }
            onChangeText={(text) =>
              props.setFieldValue('confirmPassword', text)
            }
          />
        </PaddedView>
        <Footer>
          <PaddedView>
            <Button
              disabled={props.isSubmitting}
              onPress={props.handleSubmit}
              title="Save"
              loading={props.isSubmitting}
            />
          </PaddedView>
        </Footer>
      </FlexedView>
    </StyledKeyboardAvoidingView>
  );

  render() {
    return (
      <Mutation
        mutation={CHANGE_PASSWORD}
        update={async (cache, { data: { token } }) => {
          await cache.writeQuery({
            query: GET_AUTH_TOKEN,
            data: { authToken: token },
          });

          await cache.writeData({
            data: { authToken: token },
          });
        }}
      >
        {(updatePassword) => (
          <Formik
            validationSchema={Yup.object().shape({
              password: Yup.string()
                .min(6, 'Password must be at least 6 characters')
                .required('Password is required'),
              confirmPassword: Yup.string()
                .min(6, 'Password must be at least 6 characters')
                .oneOf([Yup.ref('password')], 'Passwords must match.')
                .required('Password confirm is required'),
            })}
            onSubmit={async (variables, { setSubmitting, setFieldError }) => {
              try {
                await updatePassword({ variables });

                await this.props.navigation.goBack();
              } catch (e) {
                const { graphQLErrors } = e;
                if (graphQLErrors.length) {
                  setFieldError(
                    'confirmPassword',
                    'Unknown error. Please try again later.'
                  );
                }
              }
              setSubmitting(false);
            }}
          >
            {this.renderForm}
          </Formik>
        )}
      </Mutation>
    );
  }
}

export default ChangePassword;
