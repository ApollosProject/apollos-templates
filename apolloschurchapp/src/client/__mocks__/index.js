import React from 'react';
import { MockedProvider } from 'react-apollo/test-utils';
import ApolloClient from 'apollo-client';
import { makeExecutableSchema, addMockFunctionsToSchema } from 'graphql-tools';
import { SchemaLink } from 'apollo-link-schema';
import { importSchema } from 'graphql-import';

import cache from '../cache';
import { resolvers, schema, defaults } from '../../store';

// eslint-disable-next-line
export default (props) => {
  let finalProps = props;
  // eslint-disable-next-line
  if (!props.mocks) {
    // eslint-disable-next-line
    finalProps = { ...props, resolvers };
  }
  return <MockedProvider cache={cache} {...finalProps} />;
};

const serverSchema = makeExecutableSchema({
  typeDefs: importSchema(
    `${process.cwd()}/../apollos-church-api/local.graphql`
  ),
  resolverValidationOptions: {
    requireResolversForResolveType: false,
  },
});
addMockFunctionsToSchema({ schema: serverSchema });

const link = new SchemaLink({ schema: serverSchema });
cache.writeData({ data: defaults });

export const client = new ApolloClient({
  link,
  cache,
  resolvers,
  typeDefs: schema,
});
