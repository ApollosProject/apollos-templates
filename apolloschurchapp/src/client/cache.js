import { InMemoryCache } from '@apollo/client/cache';
import AsyncStorage from '@react-native-community/async-storage';
import { CachePersistor } from 'apollo3-cache-persist';
import ApollosConfig from '@apollosproject/config';

import introspectionQueryResultData from './fragmentTypes.json';

const finalPossibleTypes = {};
introspectionQueryResultData.__schema.types.forEach((supertype) => {
  if (supertype.possibleTypes) {
    finalPossibleTypes[supertype.name] = [
      ...supertype.possibleTypes.map((subtype) => subtype.name),
    ];
  }
});
// We reset our apollo cache based an env value and static number.
// In the future, we should also look at resetting the app when an error occurs related to Apollo.
// You can also increment this number to force a manual reset of the cache.
const SCHEMA_VERSION = `${ApollosConfig.SCHEMA_VERSION}-1`; // Must be a string.
const SCHEMA_VERSION_KEY = 'apollo-schema-version';

const nodeCacheRedirect = (_, { id }, { getCacheKey }) =>
  id ? getCacheKey({ __typename: id.split(':')[0], id }) : null;

const cache = new InMemoryCache({
  possibleTypes: finalPossibleTypes,
  cacheRedirects: {
    Query: {
      node: nodeCacheRedirect,
    },
  },
});

const persistor = new CachePersistor({
  cache,
  storage: AsyncStorage,
});

// 1. If we are on the same schema version, restore the cache.
// 2. If that fails, purge the cache and update the stored version so we don't try and restore agian
// 3. If we are on a new schema version, purge the cache.
// 4. If purging or setting AsyncStorage keys fails (or anything else) we capture the error and log it.

export const ensureCacheHydration = (async () => {
  // We wrap everything in try/catch because crashing on a cache restore is bad
  // ðŸ˜¬
  try {
    const currentVersion = await AsyncStorage.getItem(SCHEMA_VERSION_KEY);
    if (currentVersion === SCHEMA_VERSION) {
      // If the current version matches the latest version,
      // we're good to go and can restore the cache.
      try {
        await persistor.restore();
      } catch (restoreError) {
        // If the restore fails, we want to do our best to purge the cache.
        await persistor.purge();
        console.error('Error restoring cache, purging the cache', restoreError);
      }
    } else {
      // Otherwise, we'll want to purge the outdated persisted cache
      // and mark ourselves as having updated to the latest version.
      await persistor.purge();
      await AsyncStorage.setItem(SCHEMA_VERSION_KEY, SCHEMA_VERSION);
    }
  } catch (error) {
    console.error('Error restoring or purging Apollo cache', error);
  }
})();

export default cache;
