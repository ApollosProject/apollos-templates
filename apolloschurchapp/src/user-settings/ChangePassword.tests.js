import React from 'react';

import { Providers, renderWithApolloData } from '@apollosproject/ui-test-utils';
import { MockedProvider } from '@apollo/client/testing';
import ChangePassword from './ChangePassword';

describe('Change Password component', () => {
  it('renders a change password form', async () => {
    const navigation = {
      navigate: jest.fn(),
      goBack: jest.fn(),
    };
    const tree = await renderWithApolloData(
      <Providers MockedProvider={MockedProvider}>
        <ChangePassword navigation={navigation} />
      </Providers>
    );
    expect(tree).toMatchSnapshot();
  });
});
