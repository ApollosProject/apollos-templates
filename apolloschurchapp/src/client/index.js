import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import { ApolloProvider } from 'react-apollo';
import { ApolloClient } from 'apollo-client';
import { ApolloLink } from 'apollo-link';
import { getVersion, getApplicationName } from 'react-native-device-info';

import { authLink, buildErrorLink } from '@apollosproject/ui-auth';
import { resolvers, schema, defaults } from '../store';
import NavigationService from '../NavigationService';
import httpLink from './httpLink';
import cache, { ensureCacheHydration, MARK_CACHE_LOADED } from './cache';

const goToAuth = () => NavigationService.resetToAuth();
const wipeData = () => cache.writeData({ data: defaults });

let clearStore;
let storeIsResetting = false;
const onAuthError = async () => {
  if (!storeIsResetting) {
    storeIsResetting = true;
    await clearStore();
  }
  storeIsResetting = false;
  goToAuth();
};

const errorLink = buildErrorLink(onAuthError);

const link = ApolloLink.from([authLink, errorLink, httpLink]);

export const client = new ApolloClient({
  link,
  cache,
  queryDeduplication: false,
  shouldBatch: true,
  resolvers,
  typeDefs: schema,
  name: getApplicationName(),
  version: getVersion(),
});

// Hack to give auth link access to method on client;
// eslint-disable-next-line prefer-destructuring
clearStore = client.clearStore;

wipeData();
// Ensure that media player still works after logout.
client.onClearStore(() => wipeData());

class ClientProvider extends PureComponent {
  static propTypes = {
    client: PropTypes.shape({
      cache: PropTypes.shape({}),
    }),
  };

  static defaultProps = {
    client,
  };

  async componentDidMount() {
    try {
      await ensureCacheHydration;
    } catch (e) {
      throw e;
    } finally {
      client.mutate({ mutation: MARK_CACHE_LOADED });
    }
  }

  render() {
    return <ApolloProvider {...this.props} client={client} />;
  }
}

export default ClientProvider;
