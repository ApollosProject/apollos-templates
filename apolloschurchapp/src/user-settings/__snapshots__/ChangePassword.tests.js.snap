// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Change Password component renders a change password form 1`] = `
Array [
  <View
    onLayout={[Function]}
    pointerEvents="box-none"
    style={
      Array [
        Object {
          "bottom": 0,
          "left": 0,
          "overflow": "hidden",
          "position": "absolute",
          "right": 0,
          "top": 0,
        },
        Object {},
      ]
    }
  />,
  <View
    onLayout={[Function]}
    style={
      Array [
        Object {
          "backgroundColor": "#FFFFFF",
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        },
        Object {
          "paddingBottom": 0,
        },
      ]
    }
  >
    <View
      style={
        Object {
          "flex": 1,
        }
      }
    >
      <View
        style={
          Object {
            "paddingHorizontal": 16,
            "paddingVertical": 16,
          }
        }
      >
        <View
          disabled={false}
          style={
            Object {
              "marginVertical": 16,
            }
          }
        >
          <View>
            <View
              style={
                Object {
                  "flexDirection": "row",
                  "justifyContent": "center",
                }
              }
            >
              <View
                style={
                  Object {
                    "alignItems": "center",
                    "justifyContent": "center",
                  }
                }
              />
              <View
                style={
                  Object {
                    "alignSelf": "flex-end",
                    "flex": 1,
                  }
                }
              >
                <View
                  style={
                    Object {
                      "opacity": 0,
                    }
                  }
                >
                  <TextInput
                    allowFontScaling={true}
                    autoCapitalize="none"
                    autoCorrect={false}
                    editable={true}
                    onBlur={[Function]}
                    onChangeText={[Function]}
                    onFocus={[Function]}
                    placeholderTextColor="rgba(48, 48, 48, 0.6)"
                    rejectResponderTermination={true}
                    returnKeyType="done"
                    secureTextEntry={true}
                    style={
                      Array [
                        Object {
                          "color": "#303030",
                          "fontFamily": "InterUI-Bold",
                          "fontSize": 14,
                          "height": 38,
                          "paddingVertical": 0,
                        },
                        undefined,
                      ]
                    }
                    type="password"
                    underlineColorAndroid="transparent"
                  />
                </View>
                <View
                  onLayout={[Function]}
                  pointerEvents="none"
                  style={
                    Object {
                      "bottom": 0,
                      "justifyContent": "center",
                      "opacity": 1,
                      "position": "absolute",
                      "top": 0,
                      "transform": Array [
                        Object {
                          "scale": 1,
                        },
                        Object {
                          "translateX": 0,
                        },
                        Object {
                          "translateY": 0,
                        },
                      ],
                    }
                  }
                >
                  <Text
                    style={
                      Object {
                        "backgroundColor": "transparent",
                        "color": "#505050",
                        "fontFamily": "InterUI-Medium",
                        "fontSize": 14,
                        "lineHeight": 21,
                        "paddingVertical": 4,
                      }
                    }
                  >
                    New Password
                  </Text>
                </View>
              </View>
              <View
                style={
                  Object {
                    "alignItems": "center",
                    "justifyContent": "center",
                  }
                }
              />
            </View>
            <View
              style={
                Object {
                  "bottom": 0,
                  "left": 0,
                  "right": 0,
                }
              }
            >
              <View
                style={
                  Object {
                    "backgroundColor": "#A5A5A5",
                    "bottom": -0.5,
                    "height": 0.5,
                  }
                }
              />
              <View
                style={
                  Object {
                    "bottom": 0,
                    "left": 0,
                    "right": 0,
                    "transform": Array [
                      Object {
                        "scaleX": 0.0001,
                      },
                    ],
                  }
                }
              >
                <View
                  hasError={false}
                  style={
                    Object {
                      "backgroundColor": "#17B582",
                      "bottom": 0,
                      "height": 0.5,
                    }
                  }
                />
              </View>
            </View>
          </View>
        </View>
        <View
          disabled={false}
          style={
            Object {
              "marginVertical": 16,
            }
          }
        >
          <View>
            <View
              style={
                Object {
                  "flexDirection": "row",
                  "justifyContent": "center",
                }
              }
            >
              <View
                style={
                  Object {
                    "alignItems": "center",
                    "justifyContent": "center",
                  }
                }
              />
              <View
                style={
                  Object {
                    "alignSelf": "flex-end",
                    "flex": 1,
                  }
                }
              >
                <View
                  style={
                    Object {
                      "opacity": 0,
                    }
                  }
                >
                  <TextInput
                    allowFontScaling={true}
                    autoCapitalize="none"
                    autoCorrect={false}
                    editable={true}
                    onBlur={[Function]}
                    onChangeText={[Function]}
                    onFocus={[Function]}
                    placeholderTextColor="rgba(48, 48, 48, 0.6)"
                    rejectResponderTermination={true}
                    returnKeyType="done"
                    secureTextEntry={true}
                    style={
                      Array [
                        Object {
                          "color": "#303030",
                          "fontFamily": "InterUI-Bold",
                          "fontSize": 14,
                          "height": 38,
                          "paddingVertical": 0,
                        },
                        undefined,
                      ]
                    }
                    type="password"
                    underlineColorAndroid="transparent"
                  />
                </View>
                <View
                  onLayout={[Function]}
                  pointerEvents="none"
                  style={
                    Object {
                      "bottom": 0,
                      "justifyContent": "center",
                      "opacity": 1,
                      "position": "absolute",
                      "top": 0,
                      "transform": Array [
                        Object {
                          "scale": 1,
                        },
                        Object {
                          "translateX": 0,
                        },
                        Object {
                          "translateY": 0,
                        },
                      ],
                    }
                  }
                >
                  <Text
                    style={
                      Object {
                        "backgroundColor": "transparent",
                        "color": "#505050",
                        "fontFamily": "InterUI-Medium",
                        "fontSize": 14,
                        "lineHeight": 21,
                        "paddingVertical": 4,
                      }
                    }
                  >
                    Confirm Password
                  </Text>
                </View>
              </View>
              <View
                style={
                  Object {
                    "alignItems": "center",
                    "justifyContent": "center",
                  }
                }
              />
            </View>
            <View
              style={
                Object {
                  "bottom": 0,
                  "left": 0,
                  "right": 0,
                }
              }
            >
              <View
                style={
                  Object {
                    "backgroundColor": "#A5A5A5",
                    "bottom": -0.5,
                    "height": 0.5,
                  }
                }
              />
              <View
                style={
                  Object {
                    "bottom": 0,
                    "left": 0,
                    "right": 0,
                    "transform": Array [
                      Object {
                        "scaleX": 0.0001,
                      },
                    ],
                  }
                }
              >
                <View
                  hasError={false}
                  style={
                    Object {
                      "backgroundColor": "#17B582",
                      "bottom": 0,
                      "height": 0.5,
                    }
                  }
                />
              </View>
            </View>
          </View>
        </View>
      </View>
      <RCTSafeAreaView
        emulateUnlessSupported={true}
        style={
          Object {
            "flex": 1,
            "justifyContent": "flex-end",
          }
        }
      >
        <View
          style={
            Object {
              "paddingHorizontal": 16,
              "paddingVertical": 16,
            }
          }
        >
          <View
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "opacity": 1,
              }
            }
          >
            <View
              bordered={false}
              disabled={false}
              pill={true}
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#00676D",
                  "borderColor": "#00676D",
                  "borderRadius": 48,
                  "borderWidth": 2,
                  "elevation": 2,
                  "flexDirection": "row",
                  "height": 48,
                  "justifyContent": "center",
                  "opacity": 1,
                  "overflow": "hidden",
                  "paddingHorizontal": 16,
                }
              }
            >
              <Text
                style={
                  Object {
                    "color": "#FFFFFF",
                    "fontFamily": "InterUI-Bold",
                    "fontSize": 16,
                    "lineHeight": 24,
                  }
                }
              >
                Save
              </Text>
            </View>
          </View>
        </View>
      </RCTSafeAreaView>
    </View>
  </View>,
]
`;
